client/driver/docker.go

```diff
@@ -119,13 +107,10 @@ func NewDockerDriverConfig(task *structs.Task) (*DockerDriverConfig, error) {
 	if err := mapstructure.WeakDecode(task.Config, &driverConfig); err != nil {
 		return nil, err
 	}
-	if strings.Contains(driverConfig.ImageName, "https://") {
```

Comment:
So this is when the image repo is `https` based, and users just copy the url of the image. Since we default `driverConfig.SSL` to `true`, docker uses the appropriate mechanism to pull down the image.